tools/ack-discover/controller.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import dataclasses
+import os
+
+import github
+
+import ecrpublic
+import maintenance_phases
+import project_stages
+import service
+
+
+@dataclasses.dataclass
+class Release:
+    controller_version: str = None
+    chart_version: str = None
+    ack_runtime_version: str = None
+    aws_sdk_go_version: str = None
+
+
+@dataclasses.dataclass
+class Controller:
+    service: service.Service
+    latest_release: Release
+    project_stage: str = project_stages.NONE
+    maintenance_phase: str = maintenance_phases.NONE
+    source_repo_url: str = None
+    image_repo: str = None
+    chart_repo: str = None
+
+
+def collect_all(writer, gh, ep_client, services):
+    """Returns a map, keyed by service package name, of ControllerInfo objects
+    describing the ACK controllers.
+    """
+    writer.debug("[collect_controllers] collecting ACK controller information ... ")
+    ack_org = gh.get_organization("aws-controllers-k8s")
```

Comment:
@RedbackThomson see here... reference comment below about getting list of controllers...